### PR TITLE
BUG make sure to not munge prompt in zsh

### LINKED ISSUE #50: Prompt munging and verbosity bugs in gfortran activate and deactivate scripts.
Issue:

If I install gfortran_osx-64 from conda-forge on my OSX machine running zsh I am seeing two unwanted effects that are likely related bugs in the activate  and deactivate scripts.

- My prompt is being changed to 

`x86_64-apple-darwin13:~`

I see this is not happening on bash, only on zsh.

I have set: PROMPT="%m:%1~ %B%#%b "

So it looks like - for some reason the machine name is being munged here.

- On deactivation the compiler is emitting INFO information even though CONDA_BUILD is not set to 1.  I believe at least this 2nd part is likely related to this old bug which you may have copied over from defaults: https://github.com/conda/conda/issues/6945. Note this issue occurs in both bash and zsh.

<br/>
Environment (<code>conda list</code>):
<details>

```
x86_64-apple-darwin13:~ % conda list
# packages in environment at /Users/walter/.conda/envs/test-cww:
#
# Name                    Version                   Build  Channel
gfortran_impl_osx-64      7.3.0                hf4212f2_2    conda-forge
gfortran_osx-64           7.3.0                h22b1bf0_2    conda-forge
gmp                       6.2.0                h4a8c4bd_2    conda-forge
isl                       0.19                          0    conda-forge
ld64                      450.3                h3c32e8a_4    conda-forge
libcxx                    10.0.0               h1af66ff_2    conda-forge
libgfortran               4.0.0                         2    conda-forge
libiconv                  1.15              h0b31af3_1006    conda-forge
libllvm9                  9.0.1                h7475705_1    conda-forge
llvm-openmp               10.0.0               h28b9765_0    conda-forge
mpc                       1.1.0             h4160ff4_1007    conda-forge
mpfr                      4.0.2                h65ac59c_1    conda-forge
tapi                      1000.10.8            ha1b3eb9_4    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
x86_64-apple-darwin13:~ % conda info

     active environment : test-cww
    active env location : /Users/walter/.conda/envs/test-cww
            shell level : 2
       user config file : /Users/walter/.condarc
 populated config files : /Users/walter/.condarc
          conda version : 4.7.12
    conda-build version : not installed
         python version : 3.7.4.final.0
       virtual packages : 
       base environment : /cvmfs/sw.lsst.eu/darwin-x86_64/lsst_distrib/w_2020_09/python/miniconda3-4.7.12  (read only)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /cvmfs/sw.lsst.eu/darwin-x86_64/lsst_distrib/w_2020_09/python/miniconda3-4.7.12/pkgs
                          /Users/walter/.conda/pkgs
       envs directories : /Users/walter/.conda/envs
                          /cvmfs/sw.lsst.eu/darwin-x86_64/lsst_distrib/w_2020_09/python/miniconda3-4.7.12/envs
               platform : osx-64
             user-agent : conda/4.7.12 requests/2.22.0 CPython/3.7.4 Darwin/18.7.0 OSX/10.14.6
                UID:GID : 501:20
             netrc file : None
           offline mode : False

```
</details>

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR uses a version of the hack here (https://github.com/conda/conda/issues/7031#issuecomment-560587364) to fix #50. 

cc @ktlim  @cwwalter @erykoff

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge/clang-compiler-activation @conda-forge/core thoughts?

**Comment by beckermr** [comment]
@isuruf can you look at these osx-arm64 errors? 

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=264656&view=logs&jobId=15f8dc28-9f6f-5e3a-5b3c-2b169071e5be&j=15f8dc28-9f6f-5e3a-5b3c-2b169071e5be&t=63fa899e-c160-5185-75c4-c5fd30c90374

Do we need to rebuild something? The solver is complaining about something not being satisfied.

**Comment by erykoff** [comment]
I think the arm64 is just the canary that has failed first (due to smaller list of packages)

**Comment by beckermr** [comment]
Yup. Something is mispinned somewhere.

**Comment by cwwalter** [comment]
Rather than doing this for all cases where it is "not bash" should you be doing it explicitly for zsh or other shells that use the variable in the prompt?

**Comment by beckermr** [comment]
Sure, we can use $ZSH_NAME.

**Comment by erykoff** [comment]
I don't know if this is relevant, but currently `clang` and `llvm*` packages are available at 11.0.1, but `clang-osx-64` and `clang-osx-arm64` only go to 11(.0.0).

**Comment by beckermr** [comment]
OK my very hacking edit to the requirements seems to have fixed the issue for now. Ready for review @conda-forge/clang-compiler-activation!

File: recipe/activate-clang.sh [hunk 1]
```diff
@@ -195,6 +201,25 @@ else
     diff -U 0 -rN /tmp/old-env-$$.txt /tmp/new-env-$$.txt | tail -n +4 | grep "^-.*\|^+.*" | grep -v "CONDA_BACKUP_" | sort
     rm -f /tmp/old-env-$$.txt /tmp/new-env-$$.txt || true
   fi
+
+  # fix prompt for zsh
+  if [[ -n "${ZSH_NAME}" ]]; then
+    _conda_clang_precmd() {
+      CONDA_CLANG_OLDHOST="${HOST}"
+      HOST="${CONDA_CLANG_HOSTNAME}"
```
Comment (isuruf) [inline recipe/activate-clang.sh:209]:
```suggestion
      HOST="${CONDA_BACKUP_HOST}"
```

Comment (beckermr) [reply recipe/activate-clang.sh]: We can remove the other bits of the activater script this way. Good catch!

File: recipe/activate-clang.sh [hunk 2]
```diff
@@ -152,6 +152,12 @@ if [ "@CONDA_BUILD_CROSS_COMPILATION@" = "1" ]; then
   echo "endian = 'little'" >> $BUILD_PREFIX/meson_cross_file.txt
 fi
 
+if [[ -n "${ZSH_NAME}" ]]; then
+  _tc_activation \
+    activate @CHOST@- \
+    "CONDA_CLANG_HOSTNAME,$(hostname)"
+fi
+
```
Comment (beckermr) [inline recipe/activate-clang.sh:160]:
```suggestion
```

File: recipe/deactivate-clang.sh [hunk 3]
```diff
@@ -143,6 +143,12 @@ _tc_activation \
   "build_alias,@CBUILD@" \
   "BUILD,@CBUILD@"
 
+if [[ -n "${ZSH_NAME}" ]]; then
+  _tc_activation \
+    deactivate @CHOST@- \
+    "CONDA_CLANG_HOSTNAME,$(hostname)"
+fi
+
```
Comment (beckermr) [inline recipe/deactivate-clang.sh:151]:
```suggestion
```

File: recipe/activate-clang.sh [hunk 4]
```diff
@@ -195,6 +195,25 @@ else
...
+    _conda_clang_precmd() {
+      CONDA_CLANG_OLDHOST="${HOST}"
+      HOST="${CONDA_BACKUP_HOST}"
+    }
+
+    [[ -z \$precmd_functions ]] && precmd_functions=()
+    precmd_functions=(\$precmd_functions _conda_clang_precmd)
+
+    _conda_clang_preexec() {
+      HOST="${CONDA_CLANG_OLDHOST}"
```
Comment (isuruf) [inline recipe/activate-clang.sh:210]: Let's add a new variable `CONDA_TOOLCHAIN_HOST` and use it in our recipes and also do the following change.
```suggestion
      HOST="${CONDA_TOOLCHAIN_HOST}"
```

Comment (beckermr) [reply recipe/activate-clang.sh:210]: Sure!

Comment (beckermr) [reply recipe/activate-clang.sh]: done.

File: recipe/activate-clang.sh [hunk 5]
```diff
@@ -154,6 +154,7 @@ fi
 
 _tc_activation \
   activate @CHOST@- "HOST,@CHOST@" \
+  "CONDA_TOOLCHAIN_HOST,@CHOST@" \
```
Comment (isuruf) [inline recipe/activate-clang.sh:157]:
```suggestion
  "CONDA_TOOLCHAIN_HOST,@CHOST@" \
  "CONDA_TOOLCHAIN_BUILD,@CBUILD@" \
```

Comment (beckermr) [reply recipe/activate-clang.sh:157]: This needs to be in the deactivate script too.

File: recipe/deactivate-clang.sh [hunk 6]
```diff
@@ -118,6 +118,7 @@ fi
 
 _tc_activation \
   deactivate @CHOST@- "HOST,@CHOST@" \
+  "CONDA_TOOLCHAIN_HOST,@CHOST@" \
```
Comment (beckermr) [inline recipe/deactivate-clang.sh:121]:
```suggestion
  "CONDA_TOOLCHAIN_HOST,@CHOST@" \
  "CONDA_TOOLCHAIN_BUILD,@CBUILD@" \
```

**Comment by beckermr** [comment]
Some notes about the solver errors for posterity:

 - this version of conda build does not seem to be falling back to the full repodata
 - without the hacks above, mambabuild fails too

**Comment by beckermr** [comment]
We'll want a PR on the ctng-compiler activation scripts too. I can send that today.

**Comment by beckermr** [comment]
I opened https://github.com/conda-forge/ctng-compiler-activation-feedstock/pull/39 for zsh on linux

**Comment by beckermr** [comment]
We can mark this build broken if it is buggy per the other pr

**Pull request merged**